src/khoj/processor/content/images/image_to_entries.py

```
@@ -0,0 +1,124 @@
+        file_to_text_map = dict()
+        entries: List[str] = []
+        entry_to_location_map: List[Tuple[str, str]] = []
+        for image_file in image_files:
+            try:
+                try:
+                    loader = RapidOCR()
+                    bytes = image_files[image_file]
+                    # write the image to a temporary file
+                    timestamp_now = datetime.utcnow().timestamp()
+                    timestamp_now = datetime.utcnow().timestamp()
+                    # use either png or jpg
+                    if image_file.endswith(".png"):
+                        tmp_file = f"tmp_image_file_{timestamp_now}.png"
+                    elif image_file.endswith(".jpg") or image_file.endswith(".jpeg"):
+                        tmp_file = f"tmp_image_file_{timestamp_now}.jpg"
+                    with open(f"{tmp_file}", "wb") as f:
+                        bytes = image_files[image_file]
+                        f.write(bytes)
+                    try:
+                        image_entries_per_file = []
+                        result, _ = loader(tmp_file)
+                        if result:
+                            image_entries_per_file = [text[1] for text in result]
+                    except ImportError:
+                        logger.warning(f"Unable to process file: {image_file}. This file will not be indexed.")
+                        continue
+                except ImportError:
+                    logger.warning(f"Unable to process file: {image_file}. This file will not be indexed.")
+                    continue
+                entry_to_location_map += zip(
+                    image_entries_per_file, [image_file] * len(image_entries_per_file)
+                )  # this is an indexed map of image_entries for the pdf.
+                entries.extend(image_entries_per_file)
+                file_to_text_map[image_file] = image_entries_per_file
+            except Exception as e:
```

Are so many layers of try/catch required? Do we expect the code running in the outermost try/catch to fail under normal operating conditions? If not, we should continue to raise exceptions for those to make unexpected failures in this code path apparent

---

good point! fixed.